src/components/ReportActionItem/TripDetailsView.tsx

```diff
@@ -0,0 +1,162 @@
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+import Icon from '@components/Icon';
+import MenuItemWithTopDescription from '@components/MenuItemWithTopDescription';
+import OfflineWithFeedback from '@components/OfflineWithFeedback';
+import SpacerView from '@components/SpacerView';
+import Text from '@components/Text';
+import useLocalize from '@hooks/useLocalize';
+import useStyleUtils from '@hooks/useStyleUtils';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import useWindowDimensions from '@hooks/useWindowDimensions';
+import DateUtils from '@libs/DateUtils';
+import variables from '@styles/variables';
+import * as Expensicons from '@src/components/Icon/Expensicons';
+import CONST from '@src/CONST';
+import * as ReportUtils from '@src/libs/ReportUtils';
+import * as TripReservationUtils from '@src/libs/TripReservationUtils';
+import type {Reservation, ReservationTimeDetails} from '@src/types/onyx/Transaction';
+
+type TripDetailsViewProps = {
+    /** The active tripRoomReportID, used for Onyx subscription */
+    tripRoomReportID?: string;
+
+    /** Whether we should display the horizontal rule below the component */
+    shouldShowHorizontalRule: boolean;
+};
+
+type ReservationViewProps = {
+    reservation: Reservation;
+};
+
+function ReservationView({reservation}: ReservationViewProps) {
+    const theme = useTheme();
+    const styles = useThemeStyles();
+
+    const reservationIcon = TripReservationUtils.getTripReservationIcon(reservation.type);
+
+    const formatAirportInfo = (reservationTimeDetails: ReservationTimeDetails) =>
+        `${reservationTimeDetails?.longName ? `${reservationTimeDetails?.longName} ` : ''}${reservationTimeDetails?.shortName}`;
+    const getFormattedDate = () => {
+        switch (reservation.type) {
+            case CONST.RESERVATION_TYPE.FLIGHT:
+                return DateUtils.getFormattedTransportDate(new Date(reservation.start.date));
+            case CONST.RESERVATION_TYPE.HOTEL:
+            case CONST.RESERVATION_TYPE.CAR:
+                return DateUtils.getFormattedReservationRangeDate(new Date(reservation.start.date), new Date(reservation.end.date));
+            default:
+                return DateUtils.formatToLongDateWithWeekday(new Date(reservation.start.date));
+        }
+    };
+
+    const formattedDate = getFormattedDate();
+
+    const bottomDescription = useMemo(() => {
+        const code = `${reservation.confirmations && reservation.confirmations?.length > 0 ? `${reservation.confirmations[0].value} • ` : ''}`;
+        if (reservation.type === CONST.RESERVATION_TYPE.FLIGHT) {
+            return `${code}${reservation.company?.longName} • ${reservation?.company?.shortName ?? ''} ${reservation.route?.number}`;
```

Comment:
If there's no `company` this is going to be an awkward string of spaces/dots, yeah?

Reply:
nice catch! resolved for all